gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/log/v4/model/analytics/HistogramQuery.java

```diff
@@ -18,18 +18,13 @@
 import java.time.Duration;
 import java.time.Instant;
 import java.util.List;
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Data;
+import java.util.Optional;
 
-@Data
-@Builder
-@AllArgsConstructor
-public class HistogramQuery {
-
-    private String apiId;
-    private Instant from;
-    private Instant to;
-    private Duration interval;
-    private List<Aggregation> aggregations;
-}
+public record HistogramQuery(
+    String apiId,
+    Instant from,
+    Instant to,
+    Duration interval,
+    List<Aggregation> aggregations,
+    Optional<String> query
```

Comment:
Why is it optional here and not optional in the GroupBy one ?
We could remove the optional and simply check the nullity like in the previous PR no ?